[remoteopenhab] Introduce things for the remote things and add support for HTTPS (#8909)
openhab/openhab-addons: bundles/org.openhab.binding.remoteopenhab/README.md

The Remote openHAB binding allows to communicate with remote openHAB servers.
The communication is bidirectional.
The binding on the local server listens to any item state updates on the remote server and updates accordingly the linked channel on the local server.
It transfers any item command from the local server to the remote server.
It can map any remote thing to a local thing.
Through this mapping, in your rules (local server), you can take actions based upon status updates or status changes generated by remote things and you can take actions based upon trigger events generated by the trigger channels defined in the remote thing.

> One first usage of this binding is the distribution of your home automation system on a set of openHAB servers.

Reviewer: ```suggestion
One use of this binding is to distribute your home automation system on multiple of openHAB servers.
```

> A second usage is for users having old openHAB v1 bindings running that were not migrated to openHAB v2 or openHAB v3.

Reviewer: ```suggestion
Another use is for users to interact with older versions of openHAB that may support old openHAB v1 bindings that were not migrated to openHAB v2 or openHAB v3.
```

So if your remote server has one IPv4 address and one IPv6 address, you will discover two things in the inbox.
Just choose one of the two things.

> Once a bridge thing representing a remote openHAB server is created, all things from this remote server are discovered when you scan for new things.

Reviewer: ```suggestion
Once a bridge thing representing a remote openHAB server is created, all things from this remote server will be discovered when you scan for new things.
```

## Binding Configuration

The binding has no configuration options, all configuration is done at Thing level.

## Thing Configuration

The `server` thing has the following configuration parameters:

| Parameter          | Required | Description                                                                                                       |
|--------------------|----------|-------------------------------------------------------------------------------------------------------------------|
| host               | yes      | The host name or IP address of the remote openHAB server.                                                         |

> | useHttps           | no       | Set it to true in case you want to use HTTPS to communicate with the remote openHAB server. Default is false.     |

Reviewer: ```suggestion
| useHttps           | no       | Set to true if you want to use HTTPS to communicate with the remote openHAB server. Default is false.     |
```

| port               | yes      | The HTTP port to use to communicate with the remote openHAB server. Default is 8080.                              |

> | trustedCertificate | no       | Set it to true in case you want to use HTTPS even without a valid SSL certificate provided by your remote server. |

Reviewer: ```suggestion
| trustedCertificate | no       | Set to true if you want to use HTTPS even without a valid SSL certificate provided by your remote server. |
```

| restPath           | yes      | The subpath of the REST API on the remote openHAB server. Default is /rest                                        |
| token              | no       | The token to use when the remote openHAB server is setup to require authorization to run its REST API.            |

The `thing` thing has the following configuration parameters:

| Parameter     | Required | Description                                 |
|---------------|----------|---------------------------------------------|
| thingUID      | yes      | The thing UID in the remote openHAB server. |

> | buildChannels | no       | If set to true, a trigger channel will be automatically created and linked to each trigger channel from the remote thing. Default is true. |

Reviewer: I feel that this param should be renamed to `buildTriggerChannels` since from the description this config only affects trigger channels and not all channels.

Author: Renamed

Setting the `buildChannels` parameter to false is for the main following advanced usages :

* you don't care about the trigger channels of this remote thing and you don't want the binding to create them locally,
* you want to define the trigger channels in your configuration file, and only the channels that you will finally need,
* you want to set a specific channel ID rather than using the channel ID created by the binding.

## Thing Status

The status of any `thing` thing is a mapping of the remote thing status.
A mapping is done only when the `server` bridge is ONLINE (meaning the local server is connected to the remote server).

> Please note that every remote status different from UNKNOWN, ONLINE and OFFLINE will then be considered as OFFLINE on the local server.

Reviewer: ```suggestion
Please note that every remote status other than UNKNOWN, ONLINE and OFFLINE will then be considered as OFFLINE on the local server.
```
